mw/gpio_server/tests/test_controller.cpp

```diff
@@ -14,16 +14,57 @@
 #include "communication-core/sockets/mock/mockSocket.h"
 #include "core/common/error_code.h"
 
-using ::testing::_;
-using ::testing::Return;
+class SetPinTest : public ::testing::TestWithParam<
+                        std::tuple<uint16_t, simba::gpio::Value, simba::core::ErrorCode>> {
+};
 
-TEST(GPIO_CONTROLLER, TESTS) {
+INSTANTIATE_TEST_SUITE_P(SetPinTestParameters, SetPinTest,
+    ::testing::Values(
+        std::make_tuple(12, simba::gpio::Value::ERROR, simba::core::ErrorCode::kError),
+        std::make_tuple(14, simba::gpio::Value::HIGH, simba::core::ErrorCode::kOk),
+        std::make_tuple(21, simba::gpio::Value::LOW, simba::core::ErrorCode::kOk)
+    )
+);
+
+
+TEST_P(SetPinTest, CONTROLLER_SET_PIN_VALUE_CHECK) {
   MockSocket sock_;
-  ON_CALL(sock_, Transmit(_, _, _))
-            .WillByDefault(Return(simba::core::ErrorCode::kOk));
+  auto params = GetParam();
+  uint16_t pinID = std::get<0>(params);
+  simba::gpio::Value value = std::get<1>(params);
+  simba::core::ErrorCode ExpectedErrorCode = std::get<2>(params);
+
+  if (value != simba::gpio::Value::ERROR) {
+      EXPECT_CALL(sock_, Transmit(::testing::_, ::testing::_,::testing::_))
+          .WillOnce(::testing::Return(simba::core::ErrorCode::kError))
+          .WillOnce(::testing::Return(ExpectedErrorCode));
+  }
   simba::gpio::GPIOController gpio_(&sock_);
-  EXPECT_EQ(gpio_.SetPinValue(12, simba::gpio::Value::HIGH), simba::core::ErrorCode::kOk);
-  EXPECT_EQ(gpio_.SetPinValue(14, simba::gpio::Value::ERROR), simba::core::ErrorCode::kError);
-  EXPECT_EQ(gpio_.GetPinValue(21), simba::gpio::Value::HIGH);
-  EXPECT_EQ(gpio_.Init(12), simba::core::ErrorCode::kOk);
+  EXPECT_EQ(gpio_.SetPinValue(pinID, value), simba::core::ErrorCode::kError);
+  EXPECT_EQ(gpio_.SetPinValue(pinID, value), ExpectedErrorCode);
 }
+
+//  class GetPinTest : public ::testing::TestWithParam<
+//                          std::tuple<uint16_t, simba::gpio::Value>> {
+//  };
+
+// INSTANTIATE_TEST_SUITE_P(GetPinTestParameters, GetPinTest,
+//     ::testing::Values(
+//         std::make_tuple(12, simba::gpio::Value::ERROR),
+//         std::make_tuple(14, simba::gpio::Value::HIGH),
+//         std::make_tuple(21, simba::gpio::Value::LOW)
+//     )
+// );
+
+// TEST_P(GetPinTest, CONTROLLER_GET_PIN_VALUE_CHECK) {
+//   MockSocket sock_;
+//   ON_CALL(sock_, Transmit(::testing::_, ::testing::_,::testing::_))
+//             .WillByDefault(::testing::Return(simba::core::ErrorCode::kOk));
+//   simba::gpio::GPIOController gpio_(&sock_);
+//   auto params = GetParam();
+//   uint16_t pinID = std::get<0>(params);
+//   simba::gpio::Value value = std::get<1>(params);
+  
+//   gpio_.SetPinValue(pinID, value);
+//   EXPECT_EQ(gpio_.GetPinValue(pinID), value);
+// }
```

Comment:
Dlaczego tak ?

Reply:
Idk

Reply:
testy mozna wyłączyć  lepiej to wygląda ;)